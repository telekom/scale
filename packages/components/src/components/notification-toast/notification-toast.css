/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

:host {
  /* toast */
  --width: 366px;
  --width-icon-container: 48px;
  --radius: var(--telekom-radius-standard);
  --background: var(--telekom-color-background-surface);
  --z-index: 1;
  /* FIXME:TOKEN not sure what elevation to use here */
  --box-shadow: var(--telekom-shadow-raised-standard);

  /* icon container colors */
  --background-success-icon-container: var(
    --telekom-color-functional-success-standard
  );
  --background-warning-icon-container: var(
    --telekom-color-functional-warning-standard
  );
  --background-error-icon-container: var(
    --telekom-color-functional-danger-standard
  );
  --background-informational-icon-container: var(
    --telekom-color-functional-informational-standard
  );

  /* icon container colors */
  --background-success-text-container: var(
    --telekom-color-functional-success-subtle
  );
  --background-warning-text-container: var(
    --telekom-color-functional-warning-subtle
  );
  --background-error-text-container: var(
    --telekom-color-functional-danger-subtle
  );
  --background-informational-text-container: var(
    --telekom-color-functional-informational-subtle
  );
}
.notification-toast {
  width: calc(var(--width) - var(--width-icon-container));
  opacity: 1;
  z-index: var(--z-index);
  position: fixed;
  background: var(--background);
  box-shadow: var(--box-shadow);
  box-sizing: border-box;
  border-radius: 0 var(--radius) var(--radius) 0;
  flex-direction: column;
  justify-content: space-between;
}

.notification-toast.notification-toast--story {
  position: absolute;
}

.notification-toast.notification-toast--story.notification-toast--hide {
  opacity: 0;
}

.notification-toast.notification-toast--story.notification-toast--opened {
  opacity: 1;
}

.notification-toast.notification-toast--variant-success {
  background: var(--background-success-text-container);
}

.notification-toast.notification-toast--variant-warning {
  background: var(--background-warning-text-container);
}

.notification-toast.notification-toast--variant-error {
  background: var(--background-error-text-container);
}

.notification-toast.notification-toast--variant-informational {
  background: var(--background-informational-text-container);
}

/*
  variant icon part
*/

.notification-toast__icon {
  position: absolute;
  top: 50%;
  left: 50%;
  margin: -10px 0 0 -10px;
}

.notification-toast__icon-container {
  height: 100%;
  width: var(--width-icon-container);
  position: absolute;
  left: calc(var(--width-icon-container) * -1 + 1px);
  top: 0;
  float: left;
  border-radius: var(--radius) 0 0 var(--radius);
}

.notification-toast.notification-toast--variant-success
  .notification-toast__icon-container {
  background: var(--background-success-icon-container);
}

.notification-toast.notification-toast--variant-warning
  .notification-toast__icon-container {
  background: var(--background-warning-icon-container);
}

.notification-toast.notification-toast--variant-error
  .notification-toast__icon-container {
  background: var(--background-error-icon-container);
}

.notification-toast.notification-toast--variant-informational
  .notification-toast__icon-container {
  background: var(--background-informational-icon-container);
}

/*
  text part
*/

::slotted([slot='header']) {
  margin: 0;
  padding: 3px 0 0 10px;
  font-weight: bold;
  font-size: 16px;
}
::slotted([slot='body']) {
  padding: 3px 0 0 10px;
  margin: 0;
}
::slotted([slot='link']) {
  padding: 10px 0 15px 10px;
  margin: 0;
}

.notification-toast__text-container {
  width: calc(var(--width) - calc(var(--width-icon-container) * 1));
  min-height: 33px;
  float: left;
  position: relative;
  margin: 0 0 0 1px;
  padding: 15px 0 0 0;
}

/*
  close icon part
*/

.notification-toast__button-close {
  position: absolute;
  top: 10px;
  right: 7.5px;
  color: #191919;
  border: none;
  cursor: pointer;
  margin: 0;
  padding: 0;
  background: transparent;
}

.notification-toast__button-close svg {
  height: 19px;
  width: 19px;
  padding: 7.5px 6.5px 6.5px 6.5px;
  border-radius: 20%;
  color: var(--telekom-color-text-and-icon-standard);
}

.notification-toast__button-close:hover svg {
  background-color: white;
  color: var(--telekom-color-text-and-icon-primary-hovered);
}
