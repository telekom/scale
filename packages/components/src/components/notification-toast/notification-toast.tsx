/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

import {
  Component,
  Prop,
  Method,
  h,
  State,
  Element,
  Host,
} from '@stencil/core';
import classNames from 'classnames';
import statusNote from '../../utils/status-note';

@Component({
  tag: 'scale-notification-toast',
  styleUrl: './notification-toast.css',
  shadow: true,
})
export class NotificationToast {
  /** (optional) Toast variant */
  @Prop() variant?: 'error' | 'warning' | 'success' | 'informational' =
    'informational';
  /** (optional) Toast opened */
  @Prop({ reflect: true }) opened?: boolean;
  /** (optional) Animated toast */
  @Prop() animated?: boolean = true;
  /** (optional) Alignment choose for top and bottom */
  @Prop() alignment?:
    | 'bottom-right'
    | 'bottom-left'
    | 'top-right'
    | 'top-left' = 'top-right';
  /** (optional) Toast position at the top */
  @Prop() positionVertical?: number = 12;
  /** (optional) Toast position right */
  @Prop() positionHorizontal?: number = 12;
  /** (optional) Toast auto hide */
  @Prop() autoHide?: boolean = false;
  /** (optional) Toast auto hide duration */
  @Prop() autoHideDuration?: number = 3000;
  /** (optional) Toast fade duration */
  @Prop() fadeDuration?: number = 500;
  /** (optional) Injected CSS styles */
  @Prop({ reflect: true }) styles?: string;
  /** (do not use) it is a helper prop for storybook */
  @Prop() story?: boolean;
  /** (optional) Toast state height with offset */
  @State() toastHeightWithOffset: number = 0;
  @Prop() href: string;

  @Element() element: HTMLElement;

  hideToast: boolean = false;
  alignmentVertical: string;
  alignmentHorizontal: string;

  connectedCallback() {
    statusNote({ source: this.element, type: 'warn' });
  }

  componentWillLoad() {
    const alignmentParts = this.alignment.split('-');
    this.alignmentVertical = alignmentParts[0];
    this.alignmentHorizontal = alignmentParts[1];
  }

  componentDidRender() {
    if (this.autoHide === true) {
      setTimeout(this.close, this.autoHideDuration);
    }
  }

  close = () => {
    this.hideToast = true;
    setTimeout(() => {
      this.opened = false;
    }, this.fadeDuration);
  };

  handleIcons() {
    if (this.variant) {
      switch (this.variant) {
        case 'success':
          return (
            <scale-icon-alert-success
              class="notification-toast__icon"
              size={20}
              color="#ffffff"
              selected
              aria-hidden="true"
            />
          );
        case 'informational':
          return (
            <scale-icon-alert-information
              class="notification-toast__icon"
              size={20}
              selected
              color="#ffffff"
              aria-hidden="true"
            />
          );
        case 'error':
          return (
            <scale-icon-alert-error
              class="notification-toast__icon"
              size={20}
              selected
              color="#ffffff"
              aria-hidden="true"
            />
          );
        case 'warning':
          return (
            <scale-icon-alert-error
              class="notification-toast__icon"
              color="#ffff"
              size={20}
              selected
              aria-hidden="true"
            />
          );
      }
    }
    return;
  }

  /** Toast method: open() */
  @Method()
  async open() {
    this.opened = true;
    this.hideToast = false;
  }

  render() {
    if (this.opened) {
      return (
        <Host>
          {this.styles && <style>{this.styles}</style>}
          <style>{this.transitions(this.toastHeightWithOffset)}</style>
          <style>{this.animationStyle(this.toastHeightWithOffset)}</style>

          <div
            role="alert"
            style={{ display: `${this.opened ? '' : 'none'}` }}
            class={this.getCssClassMap()}
            part={this.getBasePartMap()}
            tabindex="0"
          >
            <div class="notification-toast__icon-container">
              {this.handleIcons()}
            </div>
            <div class="notification-toast__text-container">
              <slot name="header" />
              <slot name="body" />
              <scale-link
                href={this.href}
                class="notification-toast__link"
                role="link"
              >
                <slot name="link" />
              </scale-link>
            </div>

            <button
              part="button-dismissable"
              type="button"
              class="notification-toast__button-close"
              onClick={() => this.close()}
              tabindex={0}
              aria-label="close"
              onKeyDown={(e) => {
                if (e.key === 'Enter') {
                  this.close();
                }
              }}
            >
              <scale-icon-action-circle-close />
            </button>
          </div>
        </Host>
      );
    }
  }

  transitions = (offset) => `
      @keyframes fadeIn {
        from {
          opacity: 0;
          ${this.alignmentVertical}: -${offset}px;
        }
        to {
          opacity: 1;
          ${this.alignmentVertical}: ${this.positionVertical}px;
        }
      }
  
      @keyframes fadeOut {
        from {
          opacity: 1;
          ${this.alignmentVertical}: ${this.positionVertical}px;
        }
        to {
          opacity: 0;
          ${this.alignmentVertical}: -${offset}px;
        }
      }
    `;

  animationStyle = (offset) => {
    if (this.animated) {
      return `
        .notification-toast--show {
          ${this.alignmentHorizontal}: ${this.positionHorizontal}px;
          animation: fadeIn ${this.fadeDuration / 1000}s ease-in-out;
          ${this.alignmentVertical}: ${this.positionVertical}px;
          opacity: 1;
        },
        .notification-toast--show {
          ${this.alignmentHorizontal}: ${this.positionHorizontal}px;
          animation: fadeOut ${this.fadeDuration / 1000}s ease-in-out;
          ${this.alignmentVertical}: -${offset}px;
          opacity: 0;
        }
      `;
    }
    return `
    .notification-toast--show {
      ${this.alignmentHorizontal}: ${this.positionHorizontal}px;
      ${this.alignmentVertical}: ${this.positionVertical}px;
      opacity: 1;
    },
    .notification-toast--show {
      ${this.alignmentHorizontal}: ${this.positionHorizontal}px;
      ${this.alignmentVertical}: -${offset}px;
      opacity: 0;
    }
  `;
  };

  getToastHeightWithOffset() {
    const toastHeight = this.element.shadowRoot.querySelector('.toast')
      .scrollHeight;
    this.toastHeightWithOffset = toastHeight + this.positionVertical;
  }

  getBasePartMap() {
    return this.getCssOrBasePartMap('basePart');
  }

  getCssClassMap() {
    return this.getCssOrBasePartMap('css');
  }

  getCssOrBasePartMap(mode: 'basePart' | 'css') {
    const component = 'notification-toast';
    const prefix = mode === 'basePart' ? '' : `${component}`;

    return classNames(
      mode === 'basePart' ? 'base' : component,
      this.variant && `${prefix}--variant-${this.variant}`,
      !!this.opened && `${prefix}--opened`,
      !!!this.hideToast && `${prefix}--show`,
      !!this.hideToast && `${prefix}--hide`,
      this.story && `${prefix}--story`
    );
  }
}
