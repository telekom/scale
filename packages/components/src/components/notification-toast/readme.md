# scale-notification-toast



<!-- Auto Generated Below -->


## Properties

| Property             | Attribute             | Description                                    | Type                                                           | Default           |
| -------------------- | --------------------- | ---------------------------------------------- | -------------------------------------------------------------- | ----------------- |
| `alignment`          | `alignment`           | (optional) Alignment choose for top and bottom | `"bottom-left" \| "bottom-right" \| "top-left" \| "top-right"` | `'top-right'`     |
| `animated`           | `animated`            | (optional) Animated toast                      | `boolean`                                                      | `true`            |
| `autoHide`           | `auto-hide`           | (optional) Toast auto hide                     | `boolean`                                                      | `false`           |
| `autoHideDuration`   | `auto-hide-duration`  | (optional) Toast auto hide duration            | `number`                                                       | `3000`            |
| `fadeDuration`       | `fade-duration`       | (optional) Toast fade duration                 | `number`                                                       | `500`             |
| `opened`             | `opened`              | (optional) Toast opened                        | `boolean`                                                      | `undefined`       |
| `positionHorizontal` | `position-horizontal` | (optional) Toast position right                | `number`                                                       | `12`              |
| `positionVertical`   | `position-vertical`   | (optional) Toast position at the top           | `number`                                                       | `12`              |
| `story`              | `story`               | (do not use) it is a helper prop for storybook | `boolean`                                                      | `undefined`       |
| `styles`             | `styles`              | (optional) Injected CSS styles                 | `string`                                                       | `undefined`       |
| `variant`            | `variant`             | (optional) Toast variant                       | `"error" \| "informational" \| "success" \| "warning"`         | `'informational'` |


## Methods

### `open() => Promise<void>`

Toast method: open()

#### Returns

Type: `Promise<void>`




## Dependencies

### Depends on

- [scale-icon-alert-success](../icons/alert-success)
- [scale-icon-alert-information](../icons/alert-information)
- [scale-icon-alert-error](../icons/alert-error)
- [scale-link](../link)
- [scale-icon-action-circle-close](../icons/action-circle-close)

### Graph
```mermaid
graph TD;
  scale-notification-toast --> scale-icon-alert-success
  scale-notification-toast --> scale-icon-alert-information
  scale-notification-toast --> scale-icon-alert-error
  scale-notification-toast --> scale-link
  scale-notification-toast --> scale-icon-action-circle-close
  style scale-notification-toast fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
