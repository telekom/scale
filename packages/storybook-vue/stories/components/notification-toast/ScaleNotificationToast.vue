<template>
  <div>
    <slot name="header" />
    <slot name="body" />
    <slot name="link" />
  </div>
</template>

<script>
export default {
  props: {
    variant: {
      type: 'error' | 'warning' | 'success' | 'informational',
      default: 'informational',
    },
    alignment: {
      type: 'bottom-right' | 'bottom-left' | 'top-right' | 'top-left',
      default: 'top-right',
    },
    opened: { type: Boolean, default: false },
    animated: { type: Boolean, default: true },
    positionVertical: { type: Number, default: 12 },
    positionHorizontal: { type: Number, default: 12 },
    fadeDuration: { type: Number, default: 500 },
    styles: { type: String },
  },
};
</script>
