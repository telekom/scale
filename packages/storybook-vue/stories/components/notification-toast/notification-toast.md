<div style="display: inline-flex; align-items: center; justify-content: space-between; width: 100%;">
    <h1>Notification Toast</h1>
    <img src="assets/beta.png" alt="Beta Component" />
</div>

The standard Notification Toast component can be used to issue quick and non-disruptive feedback or notifications regarding an action.

## Beta components

This component is still in the beta phase. When testing it, keep in mind that it may not have gone through all quality control measures, and it may not yet have WCAG accessibility certification. There may be changes to this component in the future.
