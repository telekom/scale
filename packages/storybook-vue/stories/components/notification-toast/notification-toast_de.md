<div style="display: inline-flex; align-items: center; justify-content: space-between; width: 100%;">
    <h1>Notification Toast</h1>
    <img src="assets/beta.png" alt="Beta Component" />
</div>

Die standardmäßige Notification Toast Komponente kann verwendet werden, um schnelle, nicht störende Rückmeldungen oder Hinweise bezüglich einer Aktion auszugeben.

## Beta-Komponente

Diese Komponente befindet sich noch im Beta-Stadium. Wenn du sie testest, bedenke, dass sie möglicherweise noch nicht alle Qualitätskontrollmaßnahmen durchlaufen hat und noch keine WCAG-Zertifizierung zur Barrierefreiheit vorliegt. In Zukunft kann es zu Änderungen an dieser Komponente kommen.
