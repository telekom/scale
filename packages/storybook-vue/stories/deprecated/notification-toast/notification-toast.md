<div style="display: inline-flex; align-items: center; justify-content: space-between; width: 100%;">
    <h1>Notification Toast</h1>
    <img src="assets/deprecated.png" alt="Deprecated Component" />
</div>

We no longer support the Notification Toast. Please use the [Notification](?path=/docs/beta-components-notification--standard) instead.
