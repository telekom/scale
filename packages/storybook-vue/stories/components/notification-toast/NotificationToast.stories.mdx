import {
  Meta,
  Story,
  ArgsTable,
  Canvas,
  Description,
} from '@storybook/addon-docs/blocks';
import ScaleNotificationToast from './ScaleNotificationToast.vue';

<Meta
  title="Beta Components/Notification Toast"
  component={ScaleNotificationToast}
  argTypes={{
    variant: {
      defaultValue: 'informational',
      options: ['error', 'warning', 'success', 'informational'],
      control: {
        type: 'select',
      },
    },
    alignment: {
      defaultValue: 'top-right',
      options: ['bottom-right', 'bottom-left', 'top-right', 'top-left'],
      control: {
        type: 'select',
      },
    },
  }}
/>

export const Template = (args, { argTypes }) => ({
  props: {
    ...ScaleNotificationToast.props,
  },
  template: `<div style="height: 100px">
    <scale-notification-toast 
      story
      :position-vertical="positionVertical"
      :position-horizontal="positionHorizontal"
      :variant="variant" 
      :alignment="alignment"
      :opened="opened"
      ><p slot="header">Informational toast headline</p>
    </scale-notification-toast>
    </div>
  `,
});

export const TemplateText = (args, { argTypes }) => ({
  props: {
    ...ScaleNotificationToast.props,
  },
  template: `<div style="height: 100px">
    <scale-notification-toast 
      story
      :variant="variant"
      :opened="opened"
      ><p slot="header">Informational toast headline</p>
      <p slot="body">Subheadline goes here.</p>
    </scale-notification-toast>
    </div>
  `,
});

export const TemplateLink = (args, { argTypes }) => ({
  props: {
    ...ScaleNotificationToast.props,
  },
  template: `<div style="height: 100px">
    <scale-notification-toast 
      story 
      :variant="variant"
      :opened="opened"
      ><p slot="header">Informational toast headline</p>
      <p slot="link">Link Text</p>
    </scale-notification-toast>
    </div>
  `,
});

<div
  style={{
    display: 'inline-flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    width: '100%',
  }}
>
  <h1>Notification Toast</h1>
  <img src="assets/beta.png" alt="Beta Component" />
</div>

## Standard

<Canvas withSource="none">
  <Story
    name="Standard"
    args={{
      opened: true,
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

<ArgsTable story="Standard" />

```html
<scale-notification-toast opened>
  <p slot="header">Informational toast headline</p>
</scale-notification-toast>
```

## Warning

<Canvas withSource="none">
  <Story
    name="Warning"
    args={{
      opened: true,
      variant: 'warning',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-toast variant="warning" opened>
  <p slot="header">Informational toast headline</p>
</scale-notification-toast>
```

## Error

<Canvas withSource="none">
  <Story
    name="Error"
    args={{
      opened: true,
      variant: 'error',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-toast variant="error" opened>
  <p slot="header">Informational toast headline</p>
</scale-notification-toast>
```

## Success

<Canvas withSource="none">
  <Story
    name="Success"
    args={{
      opened: true,
      variant: 'success',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-toast variant="success" opened>
  <p slot="header">Informational toast headline</p>
</scale-notification-toast>
```

## Informational

<Canvas withSource="none">
  <Story
    name="Informational"
    args={{
      opened: true,
      variant: 'informational',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-toast variant="informational" opened>
  <p slot="header">Informational toast headline</p>
</scale-notification-toast>
```

## With Text

<Canvas withSource="none">
  <Story
    name="With Text"
    args={{
      opened: true,
    }}
  >
    {TemplateText.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-toast>
  <p slot="header">Informational toast headline</p>
  <p slot="body">Subheadline goes here.</p>
</scale-notification-toast>
```

## With Link

<Canvas withSource="none">
  <Story
    name="With Link"
    args={{
      opened: true,
    }}
  >
    {TemplateLink.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-toast variant="success" story>
  <p slot="header">Informational toast headline</p>
  <p slot="link">Link Text</p>
</scale-notification-toast>
```
