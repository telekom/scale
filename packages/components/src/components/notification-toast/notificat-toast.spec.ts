/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

import { newSpecPage } from '@stencil/core/testing';
import { NotificationToast } from './notification-toast';

describe('NotificationToast ', () => {
  let element;

  beforeEach(async () => {
    element = new NotificationToast();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [NotificationToast],
      html: `<scale-notification-toast opened="true"></scale-notification-toast>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.variant = 'warning';
    expect(element.getCssClassMap()).toContain('variant-warning');
  });

  it('handle variant prop', async () => {
    const page = await newSpecPage({
      components: [NotificationToast],
      html: `<scale-notification-toast opened="true">Label</scale-notification-toast>`,
    });
    page.root.variant = 'error';
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
    page.root.variant = 'warning';
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
    page.root.variant = 'informational';
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
    page.root.variant = 'success';
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });
});
