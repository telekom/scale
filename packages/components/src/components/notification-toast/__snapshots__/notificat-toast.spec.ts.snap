// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationToast  handle variant prop 1`] = `
<scale-notification-toast opened="">
  <mock:shadow-root>
    <style>
      @keyframes fadeIn {
        from {
          opacity: 0;
          top: -0px;
        }
        to {
          opacity: 1;
          top: 12px;
        }
      }
  
      @keyframes fadeOut {
        from {
          opacity: 1;
          top: 12px;
        }
        to {
          opacity: 0;
          top: -0px;
        }
      }
    </style>
    <style>
      .notification-toast--show {
          right: 12px;
          animation: fadeIn 0.5s ease-in-out;
          top: 12px;
          opacity: 1;
        },
        .notification-toast--show {
          right: 12px;
          animation: fadeOut 0.5s ease-in-out;
          top: -0px;
          opacity: 0;
        }
    </style>
    <div class="notification-toast notification-toast--opened notification-toast--show notification-toast--variant-error" part="base --variant-error --opened --show" role="alert" tabindex="0">
      <div class="notification-toast__icon-container">
        <scale-icon-alert-error aria-hidden="true" class="notification-toast__icon" color="#ffffff" selected="" size="20"></scale-icon-alert-error>
      </div>
      <div class="notification-toast__text-container">
        <slot name="header"></slot>
        <slot name="body"></slot>
        <scale-link class="notification-toast__link" role="link">
          <slot name="link"></slot>
        </scale-link>
      </div>
      <button aria-label="close" class="notification-toast__button-close" part="button-dismissable" tabindex="0" type="button">
        <scale-icon-action-circle-close></scale-icon-action-circle-close>
      </button>
    </div>
  </mock:shadow-root>
  Label
</scale-notification-toast>
`;

exports[`NotificationToast  handle variant prop 2`] = `
<scale-notification-toast opened="">
  <mock:shadow-root>
    <style>
      @keyframes fadeIn {
        from {
          opacity: 0;
          top: -0px;
        }
        to {
          opacity: 1;
          top: 12px;
        }
      }
  
      @keyframes fadeOut {
        from {
          opacity: 1;
          top: 12px;
        }
        to {
          opacity: 0;
          top: -0px;
        }
      }
    </style>
    <style>
      .notification-toast--show {
          right: 12px;
          animation: fadeIn 0.5s ease-in-out;
          top: 12px;
          opacity: 1;
        },
        .notification-toast--show {
          right: 12px;
          animation: fadeOut 0.5s ease-in-out;
          top: -0px;
          opacity: 0;
        }
    </style>
    <div class="notification-toast notification-toast--opened notification-toast--show notification-toast--variant-warning" part="base --variant-warning --opened --show" role="alert" tabindex="0">
      <div class="notification-toast__icon-container">
        <scale-icon-alert-warning aria-hidden="true" class="notification-toast__icon" color="#ffff" selected="" size="20"></scale-icon-alert-warning>
      </div>
      <div class="notification-toast__text-container">
        <slot name="header"></slot>
        <slot name="body"></slot>
        <scale-link class="notification-toast__link" role="link">
          <slot name="link"></slot>
        </scale-link>
      </div>
      <button aria-label="close" class="notification-toast__button-close" part="button-dismissable" tabindex="0" type="button">
        <scale-icon-action-circle-close></scale-icon-action-circle-close>
      </button>
    </div>
  </mock:shadow-root>
  Label
</scale-notification-toast>
`;

exports[`NotificationToast  handle variant prop 3`] = `
<scale-notification-toast opened="">
  <mock:shadow-root>
    <style>
      @keyframes fadeIn {
        from {
          opacity: 0;
          top: -0px;
        }
        to {
          opacity: 1;
          top: 12px;
        }
      }
  
      @keyframes fadeOut {
        from {
          opacity: 1;
          top: 12px;
        }
        to {
          opacity: 0;
          top: -0px;
        }
      }
    </style>
    <style>
      .notification-toast--show {
          right: 12px;
          animation: fadeIn 0.5s ease-in-out;
          top: 12px;
          opacity: 1;
        },
        .notification-toast--show {
          right: 12px;
          animation: fadeOut 0.5s ease-in-out;
          top: -0px;
          opacity: 0;
        }
    </style>
    <div class="notification-toast notification-toast--opened notification-toast--show notification-toast--variant-informational" part="base --variant-informational --opened --show" role="alert" tabindex="0">
      <div class="notification-toast__icon-container">
        <scale-icon-alert-information aria-hidden="true" class="notification-toast__icon" color="#ffffff" selected="" size="20"></scale-icon-alert-information>
      </div>
      <div class="notification-toast__text-container">
        <slot name="header"></slot>
        <slot name="body"></slot>
        <scale-link class="notification-toast__link" role="link">
          <slot name="link"></slot>
        </scale-link>
      </div>
      <button aria-label="close" class="notification-toast__button-close" part="button-dismissable" tabindex="0" type="button">
        <scale-icon-action-circle-close></scale-icon-action-circle-close>
      </button>
    </div>
  </mock:shadow-root>
  Label
</scale-notification-toast>
`;

exports[`NotificationToast  handle variant prop 4`] = `
<scale-notification-toast opened="">
  <mock:shadow-root>
    <style>
      @keyframes fadeIn {
        from {
          opacity: 0;
          top: -0px;
        }
        to {
          opacity: 1;
          top: 12px;
        }
      }
  
      @keyframes fadeOut {
        from {
          opacity: 1;
          top: 12px;
        }
        to {
          opacity: 0;
          top: -0px;
        }
      }
    </style>
    <style>
      .notification-toast--show {
          right: 12px;
          animation: fadeIn 0.5s ease-in-out;
          top: 12px;
          opacity: 1;
        },
        .notification-toast--show {
          right: 12px;
          animation: fadeOut 0.5s ease-in-out;
          top: -0px;
          opacity: 0;
        }
    </style>
    <div class="notification-toast notification-toast--opened notification-toast--show notification-toast--variant-success" part="base --variant-success --opened --show" role="alert" tabindex="0">
      <div class="notification-toast__icon-container">
        <scale-icon-alert-success aria-hidden="true" class="notification-toast__icon" color="#ffffff" selected="" size="20"></scale-icon-alert-success>
      </div>
      <div class="notification-toast__text-container">
        <slot name="header"></slot>
        <slot name="body"></slot>
        <scale-link class="notification-toast__link" role="link">
          <slot name="link"></slot>
        </scale-link>
      </div>
      <button aria-label="close" class="notification-toast__button-close" part="button-dismissable" tabindex="0" type="button">
        <scale-icon-action-circle-close></scale-icon-action-circle-close>
      </button>
    </div>
  </mock:shadow-root>
  Label
</scale-notification-toast>
`;

exports[`NotificationToast  should match snapshot 1`] = `
<scale-notification-toast opened="">
  <mock:shadow-root>
    <style>
      @keyframes fadeIn {
        from {
          opacity: 0;
          top: -0px;
        }
        to {
          opacity: 1;
          top: 12px;
        }
      }
  
      @keyframes fadeOut {
        from {
          opacity: 1;
          top: 12px;
        }
        to {
          opacity: 0;
          top: -0px;
        }
      }
    </style>
    <style>
      .notification-toast--show {
          right: 12px;
          animation: fadeIn 0.5s ease-in-out;
          top: 12px;
          opacity: 1;
        },
        .notification-toast--show {
          right: 12px;
          animation: fadeOut 0.5s ease-in-out;
          top: -0px;
          opacity: 0;
        }
    </style>
    <div class="notification-toast notification-toast--opened notification-toast--show notification-toast--variant-informational" part="base --variant-informational --opened --show" role="alert" tabindex="0">
      <div class="notification-toast__icon-container">
        <scale-icon-alert-information aria-hidden="true" class="notification-toast__icon" color="#ffffff" selected="" size="20"></scale-icon-alert-information>
      </div>
      <div class="notification-toast__text-container">
        <slot name="header"></slot>
        <slot name="body"></slot>
        <scale-link class="notification-toast__link" role="link">
          <slot name="link"></slot>
        </scale-link>
      </div>
      <button aria-label="close" class="notification-toast__button-close" part="button-dismissable" tabindex="0" type="button">
        <scale-icon-action-circle-close></scale-icon-action-circle-close>
      </button>
    </div>
  </mock:shadow-root>
</scale-notification-toast>
`;
